src/Shared/UnitTests/TestAssemblyInfo.cs

```diff
@@ -44,11 +44,33 @@ public MSBuildTestAssemblyFixture()
 
         //  Use a project-specific temporary path
         //  This is so multiple test projects can be run in parallel without sharing the same temp directory
-        string newTempPath = Path.Combine(AppContext.BaseDirectory, "TestTemp");
+        var subdirectory = Path.GetRandomFileName();
 
-        _testEnvironment.CreateFolder(newTempPath);
+        string newTempPath = Path.Combine(Path.GetTempPath(), subdirectory);
+        var assemblyTempFolder = _testEnvironment.CreateFolder(newTempPath);
```

Comment:
Hopefully the file name constraints will always be a superset of the directory name constraints.